app/src/main/java/com/nervousfish/nervousfish/activities/WaitActivity.java

```diff
@@ -131,14 +144,32 @@ public void onNewDataReceivedEvent(final NewDataReceivedEvent event) {
         } else if (event.getClazz().equals(Contact.class)) {
             final Contact contact = (Contact) event.getData();
             ContactReceivedHelper.newContactReceived(this.database, this, contact);
-
-            //This needs to be outside of the try catch block
-            this.dataReceived = contact;
             this.goToMainActivity();
         }
     }
 
     /**
+     * Called when a new byte array is received.
+     *
+     * @param event Contains the byte array
+     */
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onNewEncryptedBytesReceivedEvent(final NewEncryptedBytesReceivedEvent event) {
+        Validate.notNull(event);
+        LOGGER.info("onNewEncryptedBytesReceivedEvent called");
+        final SecretKey password = this.encryptor.makeKeyFromPassword(Integer.toString(this.key));
+        final byte[] bytes;
+        try {
+            final String bytesAsString = new String(event.getBytes(), this.constants.getCharset());
+            final String encryptedString = this.encryptor.decryptWithPassword(bytesAsString, password);
+            bytes = encryptedString.getBytes(this.constants.getCharset());
+        } catch (final IllegalBlockSizeException | BadPaddingException e) {
+            throw new EncryptionException(e);
```

Comment:
Neither `IllegalBlockSizeException` nor `BadPaddingException` seem like encryption exceptions to me 🤔